types/mustache/index.d.ts

```diff
@@ -338,6 +340,14 @@ declare class MustacheWriter {
     rawValue(token: string[]): string;
 }
 
+type TemplateSpanType = 'text' | 'name' | '#' | '&' | '^' | '!' | '>' | '=';
```

Comment:
Would it make sense to turn these string literals into type aliases or an enum that uses the "domain language" of mustache (`section | inverted | partial` etc..)? That would make sense when seeing the below in the source code at least.

```js
if (symbol === '#') value = this.renderSection(token, context, partials, originalTemplate);
else if (symbol === '^') value = this.renderInverted(token, context, partials, originalTemplate);
else if (symbol === '>') value = this.renderPartial(token, context, partials, tags);
else if (symbol === '&') value = this.unescapedValue(token, context);
else if (symbol === 'name') value = this.escapedValue(token, context);
else if (symbol === 'text') value = this.rawValue(token);
```
[janl/mustache.js mustache.mjs#L560-565](https://github.com/janl/mustache.js/blob/aca97b82c80e8fd1d36162e05e4b289380965d96/mustache.mjs#L560-L565)

Reply:
`enum` didn't work because the function returns string literals instead of enum values, but we can add aliases for the variants of the type.

```typescript
type RAW_VALUE = "text"
type ESCAPED_VALUE = "name"
type UNESCAPED_VALUE = "&"
type SECTION = "#"
type INVERTED = "^"
type COMMENT = "!"
type PARTIAL = ">"
type EQUAL = "="

type TemplateSpanType =
  | RAW_VALUE
  | ESCAPED_VALUE
  | SECTION
  | UNESCAPED_VALUE
  | INVERTED
  | COMMENT
  | PARTIAL
  | EQUAL

type TemplateSpans = Array<
  | [TemplateSpanType, string, number, number]
  | [TemplateSpanType, string, number, number, TemplateSpans, number]
  | [TemplateSpanType, string, number, number, string, number, boolean]
>
```

Reply:
Nice 👍